File: 125.valid-palindrome.js
/*
 * @lc app=leetcode id=125 lang=javascript
 *
 * [125] Valid Palindrome
 */
/**
 * @param {string} s
 * @return {boolean}
 */
var isPalindrome = function(s) {
	// let sArray = s.replace(',', '').replace(' ', '').toLowerCase().split('');
	// except for
	let sClean = s.toLowerCase().replace(/[^a-z0-9]/g, '');
	console.log(sClean);
	let l = 0,
		r = sClean.length - 1;

	// console.log(sClean.slice(mid));
	// let behind = sClean.slice(mid);
	// let front = sClean.slice(0, mid);
	// console.log(sClean.slice(0, mid));
	// console.log(behind.slice(1));

	while (l < r) {
		if (sClean.charAt(l) === sClean.charAt(r)) {
			l++;
			r--;
		}
		else return false;
	}
	return true;
};

isPalindrome('A man, a plan, 9a canal: Panama');

File: unsorted/java/125.valid-palindrome.java
/*
 * @lc app=leetcode id=125 lang=java
 *
 * [125] Valid Palindrome
 */

// @lc code=start
class Solution {
    public boolean isPalindrome(String s) {
        String t = s.replaceAll("[^a-zA-Z0-9]","").toLowerCase();
        char[] arr = t.toCharArray();
        int len =  arr.length;
        for(int i=0; i<len/2;i++){
            if(arr[i] != arr[len-i-1]){
                return false;
            }
        }
        return true;
    }
}
// @lc code=end

